pkg/kbuffer/throttler.go

```diff
@@ -0,0 +1,101 @@
+package kbuffer
```

Comment:
On the whole Throttler:

Would it be feasible to enhance the Breaker that's already used in the queue-proxy to achieve what we want? There are a few things that are different though:

1. It returns 503 once at capacity (we essentially need an unbounded queue)
2. It has a fixed capacity whereas we need a flexible one for the Throttler

It could be valuable to investigate though since that'd reuse stabilized/tested code and overall makes for less code. It's okay to abandon that though if the changes needed a too invasive and a rewrite is quicker.